sky/global_user_state.py

```diff
@@ -408,3 +437,41 @@ def get_storage() -> List[Dict[str, Any]]:
             'status': StorageStatus[status],
         })
     return records
+
+
+def get_cost_for_cluster(cluster_name: str,) -> float:
+
+    handle = get_handle_from_cluster_name(cluster_name)
+    metadata = get_cluster_metadata(cluster_name)
+
+    usage_intervals = metadata['usage_intervals']
+
+    # the case where cluster is torn-down after launch
+    if not usage_intervals:
+        start_time = get_cluster_launch_time(cluster_name)
+        query_time = int(time.time())
+        cost_from_start_to_query = get_cost_for_usage_intervals(
+            handle, [(start_time, query_time)])
+        return cost_from_start_to_query
+
+    return get_cost_for_usage_intervals(handle, usage_intervals)
+
+
+def get_cost_for_usage_intervals(
+    handle: Optional['backends.Backend.ResourceHandle'],
+    usage_intervals: List[Tuple[int, int]],
+) -> float:
+
+    if not handle:
```

Comment:
nit: code style, use `if handle is None:`
https://google.github.io/styleguide/pyguide.html#2144-decision